smartsim/_core/dispatch.py

```diff
@@ -433,22 +433,4 @@ def impl(
     return impl
 
 
-class ShellLauncher:
```

Comment:
Could we actually move the `make_shell_format_fn` to the shell launcher module as well? It's really just meant as a short hand to format `LaunchArguments` instances to input for the `ShellLauncher`

Reply:
Follow up question: did we want to move `create_job_id`, `ExecutableProtocol` and `LauncherProtocol` under some other different namespace? They're not strictly wrong here because they are mostly only used here so far, but I tend to think all three have more to do with "launching" rather than "dispatching".

Reply:
I am down! do you have any suggestions for where to move them to?

Reply:
moved to `smartsim._core.utils.launcher`